extensions/wikia/CategoryPage3/CategoryPage3.class.php

```diff
@@ -0,0 +1,194 @@
+<?php
+
+use Wikia\Template\Engine;
+use Wikia\Template\PHPEngine;
+
+class CategoryPage3 extends CategoryPageWithLayoutSelector {
+	/**
+	 * @var String - query param used for pagination
+	 */
+	private $from;
+
+	/**
+	 * @var CategoryPage3Model
+	 */
+	private $model;
+
+	/**
+	 * @var array
+	 */
+	private $trendingPages;
+
+	public function openShowCategory() {
+		parent::openShowCategory();
+
+		// Use AssetsManager for styles because it bundles all styles and blocks render so there is no FOUC
+		\Wikia::addAssetsToOutput( 'category_page3_scss' );
+	}
+
+	/**
+	 * @throws Exception
+	 */
+	public function closeShowCategory() {
+		$context = $this->getContext();
+		$request = $context->getRequest();
+		$this->from = $request->getVal( 'from' );
+
+		$this->model = new CategoryPage3Model( $context->getTitle(), $this->from );
+		$this->model->loadData();
+
+		if ( $this->model->getTotalNumberOfMembers() === 0 ) {
+			return;
+		}
+
+		$this->model->loadImages( 40, 30 );
```

Comment:
same `loadImages( 40, 30 )` is used in another place, that makes me even more convinced that these two values should be moved to consts